sklearn/preprocessing/_data.py

```diff
@@ -976,8 +979,13 @@ def partial_fit(self, X, y=None, sample_weight=None):
             constant_mask = _is_constant_feature(
                 self.var_, self.mean_, self.n_samples_seen_
             )
+            # TODO ugly hack, torch.sqrt does not support torch.float16!
```

Comment:
Yup, `torch.sqrt(x)` does not work for `torch.float16`. We could either drop support for this dtype or my solution here was to convert to float64 before calling sqrt and than converting it back to the original dtype.

Reply:
According to #12338, we should ways get `float64` `mean_` and `var_` attributes.